In [98]:
import matplotlib.pyplot as plt

In [99]:
import numpy as np

In [100]:
from sklearn.linear_model import LinearRegression

In [101]:
import pandas as pd

In [102]:
cricket_data = {'Temperature': [54.5, 59.5, 63.5, 67.5, 72.0, 78.5, 83.0],
              'Chirps': [81, 97, 103, 123, 150, 182, 195]}

In [103]:
lm = LinearRegression()

In [104]:
cricket_df = pd.DataFrame(cricket_data)

In [105]:
lm.fit(cricket_df['Temperature'].values.reshape(-1, 1), cricket_df['Chirps'])

LinearRegression()

In [106]:
coef = lm.coef_
print(coef[0])

4.254368932038836


In [107]:
cept = lm.intercept_
print(cept)

-157.816504854369


So we have our coefficient and our intercept now

Our equation is (4.25436893) * Temperature + (-157.816504854369)

def calculate_chirps(coefficient, temp, intercept):
    return (coefficient * temp) + intercept

if we knew the temperature then this equation would help calculate the chirps

In [108]:
def calculate_temperature(coefficient, chirps, intercept):
    return (chirps - intercept) / coefficient

The function above should give us a good prediction given chirps

In [109]:
print(calculate_temperature(coef, 15, cept))

[40.62094934]
